panther_lights/src/lights_driver_node.cpp

```diff
@@ -0,0 +1,149 @@
+#include <panther_lights/lights_driver_node.hpp>
+
+#include <filesystem>
+#include <memory>
+
+#include <gpiod.hpp>
+#include <ros/ros.h>
+
+#include <image_transport/image_transport.h>
+#include <sensor_msgs/image_encodings.h>
+
+#include <panther_msgs/SetLEDBrightness.h>
+
+#include <panther_lights/apa102.hpp>
+
+namespace panther_lights_driver
+{
+LightsDriverNode::LightsDriverNode(
+  const std::shared_ptr<ros::NodeHandle> private_nh, std::shared_ptr<ros::NodeHandle> nh,
+  const std::shared_ptr<image_transport::ImageTransport> it)
+: ph_(std::move(private_nh)),
+  nh_(std::move(nh)),
+  it_(std::move(it)),
+  front_panel_("/dev/spidev0.0"),
+  rear_panel_("/dev/spidev0.1")
+{
+  node_name_ = ros::this_node::getName();
+
+  const double global_brightness = ph_->param<double>("global_brightness", 1.0);
+  frame_timeout_ = ph_->param<double>("frame_timeout", 0.1);
+  num_led_ = ph_->param<int>("num_led", 46);
+
+  set_pin_value(gpiod::line::value::INACTIVE);
+
+  front_panel_ts_ = ros::Time::now();
+  rear_panel_ts_ = ros::Time::now();
+
+  front_panel_.set_global_brightness(global_brightness);
+  rear_panel_.set_global_brightness(global_brightness);
+
+  front_light_sub_ = it_->subscribe(
+    "lights/driver/front_panel_frame", 5, [this](const sensor_msgs::ImageConstPtr & msg) {
```

Comment:
```suggestion
    "lights/driver/front_panel_frame", 5, [&](const sensor_msgs::ImageConstPtr & msg) {
```
I guess this would be valid too but I'm not entirely sure which one makes more sense